src/main/java/com/smola/transport/service/reports/logic/MonthlyReportGenerator.java

```diff
@@ -0,0 +1,61 @@
+package com.smola.transport.service.reports.logic;
+
+import com.smola.transport.model.common.Distance;
+import com.smola.transport.model.common.Transit;
+import com.smola.transport.model.reports.DailyReport;
+import com.smola.transport.model.reports.DailyStatistics;
+import com.smola.transport.model.reports.MonthlyReport;
+import com.smola.transport.repository.TransitRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import javax.annotation.Resource;
+import java.math.BigDecimal;
+import java.time.LocalDate;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+@Component
+class MonthlyReportGenerator {
+    private TransitRepository transitRepository;
+    private ReportCalculator reportCalculator;
+    @Resource(name = "summaryPriceCalculator")
+    private ReportCalculator<BigDecimal, Transit> summaryPriceCalculator;
```

Comment:
You have no control over your code over there. Additional you are using reflection to inject in private field